### PR TITLE
[Refactor] Change Image Msg

### PR DESCRIPTION
**Pull request by Nathan24zz**
## Description
Use Image msg from ROS2 because of faster latency

## Type of Change

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause the existing functionality to not work as expected)

## How Has This Been Tested?

- [ ] New unit tests added.
- [x] Manual tested.

## Checklist:

- [ ] Using Branch Name Convention
    - `feature/JIRA-ID-SHORT-DESCRIPTION` if has a JIRA ticket
    - `enhancement/SHORT-DESCRIPTION` if has/has no JIRA ticket and contain enhancement
    - `hotfix/SHORT-DESCRIPTION` if the change doesn't need to be tested (urgent)
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made the documentation for the corresponding changes.

### CODE REVIEW

File: include/shisen_cpp/camera/provider/image_provider.hpp [hunk 1]
```diff
@@ -21,7 +21,10 @@
 #ifndef SHISEN_CPP__CAMERA__PROVIDER__IMAGE_PROVIDER_HPP_
 #define SHISEN_CPP__CAMERA__PROVIDER__IMAGE_PROVIDER_HPP_
 
-#include <shisen_interfaces/msg/image.hpp>
+// #include <shisen_interfaces/msg/image.hpp>
```
Comment (JayantiTA) [inline include/shisen_cpp/camera/provider/image_provider.hpp]: unused line

File: src/shisen_cpp/camera/provider/image_provider.cpp [hunk 2]
```diff
@@ -68,13 +68,8 @@ void ImageProvider::update_mat()
...
   current_mat_image = mat;
-
-  // Set image according to the compression quality
-  if (compression_quality > 0) {
-    set_image(current_mat_image.compress(compression_quality));
-  } else {
-    set_image(current_mat_image);
-  }
+  set_image(*(cv_bridge::CvImage(std_msgs::msg::Header(), "bgr8", mat)
+                  .toImageMsg()));
```
Comment (threeal) [inline src/shisen_cpp/camera/provider/image_provider.cpp]: I suggest to break this statement into several lines, e.g.:

```c++
const auto foo = make_foo(something, something_else);
set_image(foo.do_omething());
```

File: src/shisen_cpp/viewer/consumer/image_consumer.cpp [hunk 3]
```diff
@@ -25,7 +25,9 @@ namespace shisen_cpp::viewer
 
 void ImageConsumer::on_image_changed(const Image & image)
 {
-  current_mat_image = image;
+  cv_bridge::CvImagePtr cv_ptr;
+  cv_ptr = cv_bridge::toCvCopy(image, sensor_msgs::image_encodings::BGR8);
+  current_mat_image = cv_ptr->image;
```
Comment (threeal) [inline src/shisen_cpp/viewer/consumer/image_consumer.cpp]: Is it possible to just do this?
```c++
auto cv_ptr = cv_bridge::toCvCopy(...);
current_mat_image = cv_ptr->image;
```

**Comment by Nathan24zz** [comment]
For the error in GitHub workflows, do I need to Include External Repositories as shown here [https://github.com/marketplace/actions/ros-2-ci#Install-Several-Dependencies](url)?
For example:
```
- name: Build and test workspace
  uses: ichiro-its/ros2-build-and-test-action@main
  with:
    ros2-distro: foxy
    external-repos: https://github.com/ros2/common_interfaces/tree/foxy
```

File: include/shisen_cpp/camera/provider/image_provider.hpp (hunk 1 above)
Comment (Nathan24zz) [reply include/shisen_cpp/camera/provider/image_provider.hpp]: done

File: src/shisen_cpp/camera/provider/image_provider.cpp (hunk 2 above)
Comment (Nathan24zz) [reply src/shisen_cpp/camera/provider/image_provider.cpp]: ok

File: src/shisen_cpp/viewer/consumer/image_consumer.cpp (hunk 3 above)
Comment (Nathan24zz) [reply src/shisen_cpp/viewer/consumer/image_consumer.cpp]: done

**Review by threeal — DISMISSED** [review]

**Review by segara2410 — APPROVED** [review]

**Review by JayantiTA — APPROVED** [review]

**Pull request merged**